freshmaker/handlers/botas/botas_shipped_advisory.py

```diff
@@ -65,12 +70,90 @@ def handle(self, event):
+
+        index_images = self._pyxis.get_operator_indices()
+        # get latest bundle images per channel per index image filtered by higher version
+        all_bundles = self._pyxis.get_latest_bundles(index_images)
 
-        msg = "Skipping due to being blocked on further implementation for now."
+        bundles = self._pyxis.filter_bundles_by_digests(original_images_digests,
+                                                        all_bundles)
+        bundle_digests = []
+        for bundle in bundles:
+            if not bundle.get('bundle_path_digest'):
+                log.warning("Bundle %s doesn't have 'bundle_path_digests' set",
+                            bundle['bundle_path'])
+                continue
+            bundle_digests.append(bundle['bundle_path_digest'])
+        bundle_images = self._pyxis.get_images_by_digests(bundle_digests)
+
+        # now we have nvrs of bundles to rebuild
+        bundles_nvrs = self._filter_bundle_images_by_related_images(  # noqa: F841
+            bundle_images, original_images_digests)
 
         # Skip that event because we can't proceed with processing it.
         # TODO
-        # Next step would be queries Pyxis and getting the digests of these nvrs
+        # Now when we have bundle images' nvrs we can procceed with rebuilding it
+        msg = f"Skipping the rebuild of {len(bundles_nvrs)} bundle images " \
+              "due to being blocked on further implementation for now."
         db_event.transition(EventState.SKIPPED, msg)
         db.session.commit()
         return []
+
+    def _filter_bundle_images_by_related_images(self, bundle_images, digests):
+        """
+        Filter out bundle images that don't reference the original digests
+        of the operator/operand images
+
+        :param dict bundle_images: Mapping of bundle digest to list of bundle's
+        ContainerImages
+        :param set digests:  Digests of the original images that were rebuilt
+        that should be in 'related_images' and be 'pinned'
+        :return: list of NVRs of bundle images that have at least one
+        original related image that was rebuilt
+        """
+        ret_bundle_images_nvrs = set()
+        with koji_service(conf.koji_profile, log, dry_run=conf.dry_run,
+                          login=False) as session:
+            for bundle in bundle_images:
+                nvr = bundle['brew']['build']
+                build = session.get_build(nvr)
+                related_images = (
+                    build.get("build", {})
+                         .get("extra", {})
+                         .get("image", {})
+                         .get("operator_manifests", {})
+                         .get("related_images", {})
+                )
+                # skip bundle if it's not created by OSBS
+                if not related_images.get("created_by_osbs", False):
+                    continue
+
+                for pullspec in related_images.get('pullspecs', []):
```

Comment:
I don't think you actually need to check this because if `related_images.created_by_osbs` is `True`, that means that either OSBS replaced the pullspec, or the pullspec was already set to a digest. We know the digest points to the manifest list already since you performed that filtering in `filter_bundles_by_digests`. If you make this change, please also rename the method and clarify this in the docstring.

Here is the logic in OSBS that led me to determine this:
https://github.com/containerbuildsystem/atomic-reactor/blob/2c71eee0821c798c939a47e0e6ed540531e47c75/atomic_reactor/plugins/pre_pin_operator_digest.py#L94-L144

Reply:
Moving the conversation to https://github.com/redhat-exd-rebuilds/freshmaker/pull/38/files#r522335711.